### ISSUE TITLE
Nanorange failed with error C2607 on MSVC

### ISSUE THREAD

**Issue by QuellaZhang** [tcbrindle/NanoRange#113]
Hi all,

NanoRange assert failed on upcoming version of MSVC: F:\gitP\tcbrindle\nanorange\test\test_concepts.cpp(183): error C2607: static assertion failed

The failing line is:

static_assert(rng::default_initializable<const int>);

which eventually reaches:

template <typename T>
inline constexpr bool
is_default_initializable<T, std::void_t<decltype(::new T)>> = true;

'::new const int' is not valid as you cannot default initialize a 'const int'.

**Comment by tcbrindle** [comment]
Thanks for the notification, I think I reported this bug in MSVC so it's good to see it's been fixed (and also that [this comment](https://github.com/tcbrindle/NanoRange/blob/bf32251d65673fe170d602777c087786c529ead8/test/test_concepts.cpp#L179) seemed to work!).

Could you tell me the value of `_MSC_VER` in the fixed version?

**Comment by CaseyCarter** [comment]
> I think I reported this bug in MSVC so it's good to see it's been fixed

Yes, you reported it on DevCom. I saw Jon Caves fixing a very similar bug and convinced him to get this one as well so our `default_initializable` implementations will work. :tada:

> Could you tell me the value of `_MSC_VER` in the fixed version?

1929, which will be the `_MSC_VER` for both 16.10 _and_ 16.11. This will probably be in 16.10p3.

**Comment by tcbrindle** [comment]
Thanks Casey!
